Добавить новый функционал

va11ova11 opened the pull request:
1. Создать 2 хранилища для User и Film
2. Создать 2 сервисных класса,.UserService в нем добавить возможности добавления в друзья, удаление из друзей и вывод общих друзей. FilmService добавить возможности поставить удалить лайк фильму, вывести самые популярные фильмы по лайкам.

Commit:
"refactoring all app"

Commit:
"1. add annotation for validate Film field release date and SpaceValidation for user login"

Commit:
"1. create custom Exception and FilmStorage and UserStorage"

Commit:
"1. Create UserService"

Commit:
"Creat endpoints:
1. getUserByID
2. addFriend
3. deleteFriend
4. getAllFriend
5. getCommonFriend"

Commit:
"Creat endpoints:
1. implement all operations with users
2. tests in postman are successfully performed
3. deleted customValidators classes and test for them"

Commit:
"1.Creat FilmService
 2.Creat new method in FilmController"

krylov-sergey-sergeevich commented on .mvn/wrapper/maven-wrapper.properties:
```diff
@@ -0,0 +1,2 @@
+distributionUrl=https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.8.6/apache-maven-3.8.6-bin.zip
```
Обычно этот файл в гитигнор кидают

krylov-sergey-sergeevich commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,53 +1,70 @@
-      films.put(film.getId(), film);
-      log.trace("Validation has been successfully, the new movie has been successfully added.");
-      return film;
+  @Autowired
```
Давай воспользуемся аннотацией ломбока чтобы не писать конструктор во всех местах

krylov-sergey-sergeevich commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -1,30 +1,39 @@
-@Value
-@Builder(toBuilder = true)
+@Data
 public class Film {
```
Модификаторы доступа

krylov-sergey-sergeevich commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -1,26 +1,35 @@
-@Value
-@Builder(toBuilder = true)
+@Data
 public class User {
```
Модификаторы доступа

krylov-sergey-sergeevich commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -0,0 +1,86 @@
+    Film film = filmStorage.get(id);
+
+    if (!userService.containsUser(userId)) {
+      throw new NotFoundException("User not found");
```
В исключение всегда должна быть полная информация о произошедшем тут мы не указали какого человека не существует

krylov-sergey-sergeevich reviewed (changes requested):
Привет, в целом неплохо - давай поправим что указал и не забудь во всех исключениях учесть что нужно делать их подробными.

Commit:
" 1. Change User and film access modifier
  2. Add lombok RequiredArgsConstructor instead of constructor
  3. Add exception comment, and log FilmService and UserService"

krylov-sergey-sergeevich reviewed (approved):
Привет, вижу правки - зачет!
